dask/dataframe/groupby.py

```diff
@@ -141,6 +141,13 @@ def _groupby_raise_unaligned(df, **kwargs):
 def _groupby_slice_apply(df, grouper, key, func):
     # No need to use raise if unaligned here - this is only called after
     # shuffling, which makes everything aligned already
+    if isinstance(grouper, (pd.DataFrame, pd.Series, pd.Index)):
+        grouper = grouper.sort_values()
+    else:
+        try:
+            df = df.sort_values(grouper)
+        except KeyError:  # this fails when the grouper includes the index
+            pass
```

Comment:
@TomAugspurger any thoughts on this solution?

Reply:
Not especially, though I am going through pandas' groupby code today, so perhaps in a few hours.

Newer versions of pandas allow for the `by` in `df.sort_values(by)` to refer to columns or index level names. Are we sure that `except` is ever hit?

Reply:
Yes, in `test_groupby_column_and_index_apply` with a grouper `[index_name, column_name]`